compiler/rustc_session/src/options.rs

```diff
@@ -321,25 +327,28 @@ fn build_options<O: Default>(
 
         let option_to_lookup = key.replace('-', "_");
         match descrs.iter().find(|(name, ..)| *name == option_to_lookup) {
-            Some((_, setter, type_desc, _)) => {
-                if !setter(&mut op, value) {
-                    match value {
-                        None => early_error(
-                            error_format,
-                            &format!(
-                                "{0} option `{1}` requires {2} ({3} {1}=<value>)",
-                                outputname, key, type_desc, prefix
-                            ),
+            Some((_, setter, type_desc, _)) => match setter(&mut op, value) {
+                Ok(()) => {}
+                Err(ParseError::Invalid) => match value {
+                    None => early_error(
+                        error_format,
+                        &format!(
+                            "{0} option `{1}` requires {2} ({3} {1}=<value>)",
+                            outputname, key, type_desc, prefix
                         ),
-                        Some(value) => early_error(
-                            error_format,
-                            &format!(
-                                "incorrect value `{value}` for {outputname} option `{key}` - {type_desc} was expected"
-                            ),
+                    ),
+                    Some(value) => early_error(
+                        error_format,
+                        &format!(
+                            "incorrect value `{value}` for {outputname} option `{key}` - {type_desc} was expected"
                         ),
-                    }
-                }
-            }
+                    ),
+                },
+                Err(ParseError::Removed(in_favor_of)) => early_error(
+                    error_format,
+                    &format!("{outputname} option `{key}` removed in favor of `{in_favor_of}`"),
```

Comment:
Won’t this make the diagnostic here non-translatable? Do we want it to be translatable?

Reply:
It's just as untranslatable as the existing error message above it. And the inputs are just options, so I don't see why this couldn't be migrated